lib/x509/signature_algorithm.ex

```diff
@@ -15,6 +17,11 @@ defmodule X509.SignatureAlgorithm do
     new(hash, algorithm, type)
   end
 
+  def new(hash, {:ECPrivateKey, _, _, {:namedCurve, curve}, _, _}, type)
```

Comment:
```suggestion
  def new(hash, ec_private_key(parameters: {:namedCurve, curve}), type)
```

Reply:
Ah, that's much nicer - knew there was a better way about that one!